src/hotspot/share/oops/resolvedFieldEntry.hpp

```diff
@@ -102,8 +102,8 @@ class ResolvedFieldEntry {
   void print_on(outputStream* st) const;
 
   void set_flags(bool is_final, bool is_volatile) {
-    u1 new_flags = (static_cast<u1>(is_final) << static_cast<u1>(is_final_shift)) | static_cast<u1>(is_volatile);
-    _flags = new_flags;
+    int new_flags = (is_final << is_final_shift) | static_cast<int>(is_volatile);
+    _flags = (u1)new_flags;
```

Comment:
Why isn't this using checked_cast?

Reply:
I still don't see the point of using checked_cast if we know that this is only two flags.

Reply:
As far as I've designed it, only the last two bits of the flags are ever set or used. C++ booleans are either 0 or 1 so the higher bits should never be set.

Reply:
It is only two bits and the only variable here "is_final_shift" is 1.  This could use 1 to make it obvious but the variable is used in the templateInterpreter code.  Again, I think this is good.  This is never going to be a bug.

Reply:
It's fine unless the shift values get changed so the value no longer fits in u1.  Then it fails silently.
As an alternative to checked_cast, how about something like:
```
assert(is_final() == is_final, "");
assert(is_volatile() == is_volatile, "");
```